com.ibm.streamsx.topology/opt/python/packages/streamsx/scripts/sc.py

```diff
@@ -51,45 +52,95 @@ def _parse_dependencies():
             deps[name] = version
     return deps
 
-def _add_local_toolkits(toolkits, dependencies, topo):
+def _add_local_toolkits(toolkit_paths, dependencies, topo):
     """ A helper function that given the local toolkit paths, and the dependencies (ie toolkits that the apps depends on),
-    checks and adds the local toolkit if it is one of the dependencies
+    first checks whether the dependency w/ correct version already exists on the remote buildserver, and if not checks
+    whether the dependency w/ correct version exists locally, and if so adds it
 
     Arguments:
-        toolkits {List} -- A list of local toolkit directories, each could be a toolkit directory (has info.xml directly inside),
+        toolkit_paths {List} -- A list of local toolkit directories, each could be a toolkit directory (has info.xml directly inside),
                 or a directory consisting of toolkits (no info.xml directly inside)
-        dependencies {Dictionary} -- A dictionary where the key is the names of the dependencies, and the value is its version.
+        dependencies {Dictionary} -- A dictionary consisting of elements where the key is the name of the dependency, and the value is its version.
         topo {topology Object} -- [description]
     """
+    sc = BuildService.of_endpoint(verify=False)
```

Comment:
`sc` as a local variable name doesn't intuitively imply _build service_ and given this tool is **sc** the local variable could have a more appropriate name.